build/golang.go

```diff
@@ -31,3 +24,33 @@ func (d *Driver) depEnsure() error {
 	}
 	return nil
 }
+
+func goExec(dir string, args ...string) error {
```

Comment:
[optional] I sometimes find it helpful to have a command-builder:

```go
func goCmd(dir, modules string, args ...string) *exec.Cmd {
   cmd := exec.Command("go", args...)
   cmd.Stderr = os.Stderr
   cmd.Stdout = os.Stdout
   cmd.Dir = dir
   cmd.Env = append(os.Environ(), "GO111MODULE="+modules)
   return cmd
}

func goExec(dir string, args ...string) error { return goCmd(dir, "off", args...).Run() }

func goMod(dir string, args ...string) error { return goCmd(dir, "on", args...).Run() }
```

This also makes it easy to add additional bits to the command before running it, if necessary, without replicating the setup. YMMV.

Reply:
I'd love this to work, but a usual error text from `Run` is something like `"exit code 1"`. So it's necessary to catch `stderr` and replace an error text with it, for example.

But it seems like I was too lazy to catch it :man_facepalming:

Reply:
Nevermind, I will just redefine the `Run` method then.

Reply:
Ah: Note that the `Output` method handles this for you: If you call `c.Output()` the return value is the stdout, and stderr is captured for you: https://godoc.org/os/exec#ExitError

So if that's the main reason we're doing this, maybe this will help.